src/include/parser/update_statement.h

```diff
@@ -54,38 +54,29 @@ class UpdateStatement : public SQLStatement {
    * @param updates update clauses
    * @param where update conditions
    */
-  UpdateStatement(std::shared_ptr<TableRef> table, std::vector<std::shared_ptr<UpdateClause>> updates,
-                  std::shared_ptr<AbstractExpression> where)
-      : SQLStatement(StatementType::UPDATE),
-        table_(std::move(table)),
-        updates_(std::move(updates)),
-        where_(std::move(where)) {}
+  UpdateStatement(std::unique_ptr<TableRef> table, std::vector<std::unique_ptr<UpdateClause>> updates,
+                  common::ManagedPointer<AbstractExpression> where)
+      : SQLStatement(StatementType::UPDATE), table_(std::move(table)), updates_(std::move(updates)), where_(where) {}
 
   UpdateStatement() : SQLStatement(StatementType::UPDATE), table_(nullptr), where_(nullptr) {}
 
   ~UpdateStatement() override = default;
 
   void Accept(SqlNodeVisitor *v) override { v->Visit(this); }
 
-  /**
-   * @return update table target
-   */
-  std::shared_ptr<TableRef> GetUpdateTable() { return table_; }
+  /** @return update table target */
+  common::ManagedPointer<TableRef> GetUpdateTable() { return common::ManagedPointer(table_); }
 
-  /**
-   * @return update clauses
-   */
-  std::vector<std::shared_ptr<UpdateClause>> GetUpdateClauses() { return updates_; }
+  /** @return update clauses */
+  const std::vector<std::unique_ptr<UpdateClause>> &GetUpdateClauses() { return updates_; }
```

Comment:
Should we remove the const guard here? I may need to do std::move on the `updates_` when creating a logical update operator